### DISCUSSION TITLE
Assertion for private property does not work with getter

### DISCUSSION THREAD

**Discussion by tdgroot** [vimeo/psalm#5636] [Q&A]
My snippet: https://psalm.dev/r/79ceb448ff.

I would like to check if the property is not null through the `hasFoo` function and get the (not-null) value with `getFoo`.

What can I do to make Psalm understand that the returning `getFoo` value is not null?

**Comment by weirdan** [comment]
You can do that with templates (https://psalm.dev/r/d8e2d7c1cb) but it's not very convenient.

**Reply by tdgroot** [reply]
@weirdan should I create an issue as a feature request or is this issue already known?

**Reply by weirdan** [reply]
I have recently learned that Psalm can understand assertions on method returns, e.g. https://psalm.dev/r/1e559afe82.
